server/runner.go

```diff
@@ -0,0 +1,244 @@
+		ReadonlyPaths: []string{
+			"/proc/sys", "/proc/sysrq-trigger", "/proc/irq", "/proc/bus",
+		},
+		Devices:  configs.DefaultAutoCreatedDevices,
+		Hostname: "testing",
+		Mounts: []*configs.Mount{
+			{
+				Source:      "proc",
+				Destination: "/proc",
+				Device:      "proc",
+				Flags:       defaultMountFlags,
+			},
+			{
+				Source:      "tmpfs",
+				Destination: "/dev",
+				Device:      "tmpfs",
+				Flags:       unix.MS_NOSUID | unix.MS_STRICTATIME,
+				Data:        "mode=755",
+			},
+			{
+				Source:      "devpts",
+				Destination: "/dev/pts",
+				Device:      "devpts",
+				Flags:       unix.MS_NOSUID | unix.MS_NOEXEC,
+				Data:        "newinstance,ptmxmode=0666,mode=0620,gid=5",
+			},
+			{
+				Device:      "tmpfs",
+				Source:      "shm",
+				Destination: "/dev/shm",
+				Data:        "mode=1777,size=65536k",
+				Flags:       defaultMountFlags,
+			},
+			{
+				Source:      "mqueue",
+				Destination: "/dev/mqueue",
+				Device:      "mqueue",
+				Flags:       defaultMountFlags,
+			},
+			{
+				Source:      "sysfs",
+				Destination: "/sys",
+				Device:      "sysfs",
+				Flags:       defaultMountFlags | unix.MS_RDONLY,
+			},
+		},
+		UidMappings: []configs.IDMap{
+			{
+				ContainerID: 0,
+				HostID:      1000,
+				Size:        65536,
+			},
+		},
+		GidMappings: []configs.IDMap{
+			{
+				ContainerID: 0,
+				HostID:      1000,
+				Size:        65536,
+			},
+		},
+		Networks: []*configs.Network{
+			{
+				Type:    "loopback",
+				Address: "127.0.0.1/0",
+				Gateway: "localhost",
+			},
+		},
+		Rlimits: []configs.Rlimit{
+			{
+				Type: unix.RLIMIT_NOFILE,
+				Hard: uint64(1025),
+				Soft: uint64(1025),
+			},
+		},
+	}
+}
+
+func process(task *Task) *libcontainer.Process {
+	// TODO - Redirect stdout and stderr to a file
+	return &libcontainer.Process{
+		Args:   append([]string{task.Request.Command}, task.Request.Args...),
+		Env:    []string{"PATH=/bin"},
+		User:   "root",
+		Stdin:  os.Stdin,
+		Stdout: os.Stdout,
+		Stderr: os.Stderr,
+	}
+}
+
+type Runner struct {
+	client *clientv3.Client
+	id     *api.NodeID
+}
+
+func (r *Runner) Start(containerDir string, rootFs string) {
+	rootFs, err := filepath.Abs(rootFs)
+	if err != nil {
+		log.Fatalln("Error getting absolute rootfs path", err)
+	}
+
+	factory, err := libcontainer.New(containerDir, libcontainer.Cgroupfs, libcontainer.InitArgs(os.Args[0], "init"))
+	if err != nil {
+		log.Fatalln("Error creating libcontainer factory:", err)
+	}
+
+	// TODO - What does cgroupName do?
+	cfg := config(rootFs, "test")
+
+	newTasks := watchQueuedTasks(context.Background(), r.client)
+
+	for task := range newTasks {
+		if err := task.run(context.Background(), r.client, r.id); err != nil {
+			// TODO - Differentiate stolen task from other errors
+			continue
+		}
+
+		log.Println("Running task", task.Id.Uuid)
+
+		// TODO - Should the task be requeued on errors?
+		go func(runningTask *Task) {
+			container, err := factory.Create(task.Id.Uuid, cfg)
```

Comment:
This is too much code for inline functions. There's no strict guide rule to follow here, but I tend to want to split the function into pieces with clear purpose - manily for readability. This should be a separate `runTask` function (also ideally constrainted with a context accepted as a paremeter).